carts/models.py

```diff
@@ -84,6 +84,11 @@ def save(self, *args, **kwargs):
             None
         """
         # pylint: disable=no-member
+        self.product.stock_quantity -= self.quantity
+        if self.id is not None:
+            self.product.stock_quantity += CartItem.objects.get(id=self.id).quantity
```

Comment:
At this point i think it will be convenient that we create update methods within the product e.g. increase_quantity & decrease_quantity and we call them here and everywhere we manually adding or reducing the quantity.

Reply:
Arslan bhai, I have added a method `.update_product_quantity(self.quantity)`  which takes the new quantity and automatically updates the product quantity. Calling `decrease/increase_quantity` would require an extra step to determine which one to call that is why i made one method that does that step as well. Please let me know if this is alright